server/src/formatting.ts

```diff
@@ -3,56 +3,76 @@ import {
     DocumentFormattingParams,
     TextEdit,
     DocumentRangeFormattingParams,
+    Position,
+    Range,
     WorkspaceFolder
 } from 'vscode-languageserver/node';
 import {
     TextDocument
 } from 'vscode-languageserver-textdocument';
 import { PerlDocument, PerlElem, NavigatorSettings } from "./types";
-import {  nLog } from "./utils";
+import {  async_execFile, getPerlimportsProfile, nLog } from "./utils";
 import { dirname, join } from 'path';
+import Uri from 'vscode-uri';
 import { execFileSync } from 'child_process';
 import { getPerlAssetsPath } from "./assets";
 
 
 export function formatDoc(params: DocumentFormattingParams, txtDoc: TextDocument, settings: NavigatorSettings): TextEdit[] | undefined {
+    const start = Position.create( 0,0 );
+    const end = Position.create( txtDoc.lineCount, 0 );
+    const range = {start, end} as Range;
     const text = txtDoc.getText();
-    const fixedSource = perltidy(text, settings);
-    const start = { line: 0, character: 0 };
-    const end = { line: txtDoc.lineCount, character: 0 };
-    const range = {start, end};
-    if(fixedSource){
-        let edits: TextEdit = {
-            range: range,
-            newText: fixedSource
-        };
-        return [edits];
-    } else {
-        return;
-    }
-} 
 
-export function formatRange(params: DocumentRangeFormattingParams, txtDoc: TextDocument, settings: NavigatorSettings): TextEdit[] | undefined {
+    return maybeReturnEdits(range, txtDoc, settings);
+}
 
-    const start = { line: params.range.start.line, character: 0 };
+export function formatRange(params: DocumentRangeFormattingParams, txtDoc: TextDocument, settings: NavigatorSettings): TextEdit[] | undefined {
+    const start = Position.create( params.range.start.line, 0 );
     const offset = params.range.end.character > 0 ? 1 : 0;
-    const end = { line: params.range.end.line + offset, character: 0 };
-    const range = {start, end};
+    const end = Position.create(params.range.end.line + offset, 0 );
+    const range = {start, end} as Range;
```

Comment:
Similar to above, should be able to use `Range.create()` and `text` variable doesn't appear to be used